src/mesh/triangulator_interface.C

```diff
@@ -314,5 +315,67 @@ unsigned int TriangulatorInterface::total_hole_points()
   return n_hole_points;
 }
 
+void TriangulatorInterface::generate_auto_area_function(const Parallel::Communicator &comm,
+                                                        const unsigned int num_nearest_pts,
+                                                        const unsigned int power,
+                                                        const Number background_value,
+                                                        const Real  background_eff_dist)
+{
+  // Get the hole mesh of the outer boundary
+  // Holes should already be attached if applicable when this function is called
```

Comment:
That's more fragile than it needs to be, for a function for users to call.  Let the users call something (I'd still push for `set_desired_area_function(OurNewSubclass(its, args))` to set up the function, then we can call its `init()` ourselves from within `triangulate()` and not rely on them getting ordering correct.